--- a/qtjsbackend/src/v8/v8.pri
+++ b/qtjsbackend/src/v8/v8.pri
@@ -303,7 +303,7 @@ V8_LIBRARY_FILES = \
 V8_EXPERIMENTAL_LIBRARY_FILES = \
     $$V8SRC/proxy.js \
 
-v8_js2c.commands = python $$V8DIR/tools/js2c.py $$V8_GENERATED_SOURCES_DIR/libraries.cpp CORE off
+v8_js2c.commands = hostpython $$V8DIR/tools/js2c.py $$V8_GENERATED_SOURCES_DIR/libraries.cpp CORE off
 v8_js2c.commands += $$V8SRC/macros.py ${QMAKE_FILE_IN}
 v8_js2c.output = $$V8_GENERATED_SOURCES_DIR/libraries.cpp
 v8_js2c.input = V8_LIBRARY_FILES
@@ -314,7 +314,7 @@ v8_js2c.CONFIG += combine
 v8_js2c.name = generating[v8] ${QMAKE_FILE_IN}
 silent:v8_js2c.commands = @echo generating[v8] ${QMAKE_FILE_IN} && $$v8_js2c.commands
 
-v8_js2c_experimental.commands = python $$V8DIR/tools/js2c.py $$V8_GENERATED_SOURCES_DIR/experimental-libraries.cpp EXPERIMENTAL off
+v8_js2c_experimental.commands = hostpython $$V8DIR/tools/js2c.py $$V8_GENERATED_SOURCES_DIR/experimental-libraries.cpp EXPERIMENTAL off
 v8_js2c_experimental.commands += $$V8SRC/macros.py ${QMAKE_FILE_IN}
 v8_js2c_experimental.output = $$V8_GENERATED_SOURCES_DIR/experimental-libraries.cpp
 v8_js2c_experimental.input = V8_EXPERIMENTAL_LIBRARY_FILES
